### PR TITLE
[backup] ignore "History until epoch X can't verify" error

### PR DESCRIPTION
**Pull request by msmouse**
### Description
Edge case where the last txn chunk containing the target version carries a proof from the next epoch -- we can't write that epoch ending info to the DB so we didn't load it.

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4108)

### CODE REVIEW

File: storage/backup/backup-cli/src/backup_types/epoch_ending/restore.rs [hunk 1]
```diff
@@ -275,12 +275,15 @@ impl EpochHistory {
...
-            "History until epoch {} can't verify epoch {}",
-            self.epoch_endings.len(),
-            epoch,
-        );
+        if epoch > self.epoch_endings.len() as u64 {
+            // TODO(aldenhu): fix this from upper level
+            warn!(
+                epoch = epoch,
+                epoch_history_until = self.epoch_endings.len(),
+                "Epoch too new, can't verify. Don't worry, node won't be able to start if data is not compatible.",
```
Comment (sherry-x) [inline storage/backup/backup-cli/src/backup_types/epoch_ending/restore.rs:283]: "Don't worry, node won't be able to start if data is not compatible." this sentence itself is very worrisome lol. can we come up with better description? Some thing like "Will ignore / will skip"?

Comment (msmouse) [reply storage/backup/backup-cli/src/backup_types/epoch_ending/restore.rs:283]: I'll add "Ingoring".

**Review by sherry-x — APPROVED** [review]

File: storage/backup/backup-cli/src/backup_types/epoch_ending/restore.rs [hunk 2]
```diff
@@ -275,12 +275,16 @@ impl EpochHistory {
...
-            self.epoch_endings.len(),
-            epoch,
-        );
+        if epoch > self.epoch_endings.len() as u64 {
+            // TODO(aldenhu): fix this from upper level
+            warn!(
+                epoch = epoch,
+                epoch_history_until = self.epoch_endings.len(),
+                "Epoch too new, can't verify, letting it pass. Don't worry, previous chunks are \
+                verified, and node won't be able to start if this piece of data is indeed malicious.",
```
Comment (gregnazario) [inline storage/backup/backup-cli/src/backup_types/epoch_ending/restore.rs:284]: A rewrite:

```
Epoch is too new and can't be verified.  Previous chunks are verified and node won't be able to start if this data is malicious.
```

**Review by zcchahaha — APPROVED** [review]
